src/resourcemonitor/DefaultResourceMonitor.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import ResourceMonitor from '../resourcemonitor/ResourceMonitor';
+import AsyncScheduler from '../scheduler/AsyncScheduler';
+import IntervalScheduler from '../scheduler/IntervalScheduler';
+
+/** @internal */
+export default class DefaultResourceMonitor implements ResourceMonitor {
+  private readonly intervalScheduler: IntervalScheduler;
+
+  private count: number = 0;
+  private timestampMs: number = 0;
+
+  constructor(
+    private readonly logger: Logger,
```

Comment:
If builders want to override this value, how would they do it? Doesn't seem there is a way for them to pass in these values.

Reply:
No, builders can't change this configuration in the internal component. After some tests, we can have a follow-up task:
- Expose a public audio-video observer for inefficient resources.
- We can also discuss whether to make these configurable.